tests/ResponseTest.php

```diff
@@ -111,6 +113,71 @@ public function testSetContent()
         $this->assertSame('Hello World', (string) $response->getBody());
     }
 
+    public function testSetContentWithResource()
+    {
+        $response = new Response();
+
+        $this->assertSame($response, $response->setContent(fopen('php://input', 'rb')));
+        $this->assertInstanceOf(MessageBody::class, $response->getBody());
+        $this->assertInstanceOf(MessageBodyResource::class, $response->getBody());
+        $this->assertSame(0, $response->getBody()->getSize());
+    }
+
+    public function testIsStatusCode()
+    {
+        $response = new Response();
+
+        $this->assertTrue($response->isStatusCode(200));
+        $this->assertFalse($response->isStatusCode(400));
+    }
+
+    /**
+     * @expectedException        \RuntimeException
+     * @expectedExceptionMessage Could not parse response (error 1).
+     */
+    public function testParseShouldThrowRuntimeExceptionError1()
+    {
+        Response::parse('<!DCOTYPE html><html>HTML strings</html>');
+    }
+
+    /**
+     * @expectedException        \RuntimeException
+     * @expectedExceptionMessage Could not parse response (error 2).
+     */
+    public function testParseShouldThrowRuntimeExceptionError2()
+    {
+        Response::parse('HTTP/1.0 200 OK' . "\r\n" . 'Content-Length: 20' . "\r\n");
+    }
+
+    /**
+     * @expectedException        \RuntimeException
+     * @expectedExceptionMessage Could not parse response (error 3).
+     */
+    public function testParseShouldThrowRuntimeExceptionError3()
+    {
+        Response::parse('HTTP/1.0 200 OK' . "\r\n" . 'Content-Length20' . "\r\n\r\n");
+    }
+
+    public function testParseShouldReturnResponseObject()
+    {
+        $result = Response::parse('HTTP/1.0 200 OK' . "\r\n" . 'Content-Length: 20' . "\r\n\r\n");
+
+        $this->assertInstanceOf(Response::class, $result);
+        $this->assertSame(200, $result->getStatusCode());
+        $this->assertSame('1.0', $result->getProtocolVersion());
+        $this->assertSame(['Content-Length' => ['0']], $result->getHeaders());
```

Comment:
As you can see, `Content-Length` is a bad header choice here, as even though you set the header value to `20`, it's returned as `0` here as the body has not contents.

Actually the library should have thrown an exception here, when parsing a message with a Content-Length and no body. I will work on this.

In the meantime, please just use another header, such as `Content-Type: text/html`.

Reply:
Please do the same for `testParseShouldThrowRuntimeExceptionError2` and `testParseShouldThrowRuntimeExceptionError3`

Reply:
Yes, it needs to throw ```Exception``` when setting ```Content-Length```, but it's the empty body.